### PR TITLE
Add transnetv2

### PR DESCRIPTION
**Pull request by lvjian0706**
Add transnetv2 model, update predict.py, utils.py

### CODE REVIEW

File: configs/partitioners/transnetv2/transnetv2.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+INFERENCE:
+    name: 'TransNetV2_Inference_helper'
+    num_frames: 100
+    height: 27
+    width: 48
+    num_channels: 3
+    threshold: 0.5
+    output_path: output
```
Comment (huangjun12) [inline configs/partitioners/transnetv2/transnetv2.yaml]: 这个文件夹默认是存ckpt的，可以改成 inference_results

File: docs/zh-CN/model_zoo/partition/transnetv2.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## 模型简介
+
+TransNetV2是一种基于深度学习的视频切分模型，通过DDCNN V2结构进行特征学习，并加入RGB颜色直方图、视频帧相似度进行更有效的特征提取，最终获取每一帧是否是镜头边界帧的概率，从而完成视频切分。该算法效果较好，且计算高效，十分适合工业落地。
+
+![](../../../images/transnetv2.png)
+
+本代码当前仅支持模型推理，模型的训练和测试将在后续提供。
+
+详细内容请参考论文[TransNet V2: An effective deep network architecture for fast shot transition detection](https://arxiv.org/abs/2008.04838)
```
Comment (huangjun12) [inline docs/zh-CN/model_zoo/partition/transnetv2.md]: 参考文献最后写了，这里可以不用写了

File: docs/zh-CN/model_zoo/partition/transnetv2.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+## 模型测试
+
+coming soon 
+
+
+## 模型推理
+
+### 模型准备
+
+#### 下载预训练模型
```
Comment (huangjun12) [inline docs/zh-CN/model_zoo/partition/transnetv2.md]: 这两个小标题可以不用了

File: docs/zh-CN/model_zoo/partition/transnetv2.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                           --use_gpu=True \
+                           --use_tensorrt=False
+```
+
+输出示例如下:
+
+```bash
+Current video file: data/example.avi
+      Shot Boundarys: [[  0 130]]
+```
```
Comment (huangjun12) [inline docs/zh-CN/model_zoo/partition/transnetv2.md:76]: 输出路径和visualize这两个参数文档里补充说明下吧

File: paddlevideo/metrics/transnetv2_metric.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    if len(scenes) == 0:
+        return np.array([[0, len(predictions) - 1]], dtype=np.int32)
+
+    return np.array(scenes, dtype=np.int32)
+
+
+def evaluate_scenes(gt_scenes, pred_scenes, n_frames_miss_tolerance=2):
+    """
+    Adapted from: https://github.com/gyglim/shot-detection-evaluation
+    The original based on: http://imagelab.ing.unimore.it/imagelab/researchActivity.asp?idActivity=19
```
Comment (huangjun12) [inline paddlevideo/metrics/transnetv2_metric.py:44]: Users should be careful about adopting these functions in any commercial matters.

File: paddlevideo/modeling/backbones/transnetv2.py [hunk 6]
```diff
@@ -0,0 +1,578 @@
...
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as functional
+import random
+from paddle import ParamAttr
+
+from ..registry import BACKBONES
+
+MEAN = np.array([0.485, 0.456, 0.406], np.float32).reshape([1, 3, 1, 1]) * 255
+STD = np.array([0.229, 0.224, 0.225], np.float32).reshape([1, 3, 1, 1]) * 255
```
Comment (huangjun12) [inline paddlevideo/modeling/backbones/transnetv2.py]: 作为参数从配置文件里传进来？
不推荐这么定义全局变量

File: paddlevideo/modeling/framework/partitioners/transnetv2_partitioner.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        # NOTE: (shipping) when testing, the net won't call head.loss, we deal with the test processing in /paddlevideo/metrics
+        frame_sequence = data_batch[0]
+        one_hot_pred = self.forward_net(frame_sequence)
+        return one_hot_pred
+
+    def infer_step(self, data_batch):
+        """Define how the model is going to test, from input to output."""
+        frame_sequence = data_batch[0]
+        one_hot_pred = self.forward_net(frame_sequence)
+        #print(type(one_hot_pred[1]))
```
Comment (huangjun12) [inline paddlevideo/modeling/framework/partitioners/transnetv2_partitioner.py]: delete note

File: tools/utils.py [hunk 8]
```diff
@@ -512,3 +512,145 @@ def preprocess(self, input_file):
...
+            out = out.astype(np.float32)
+            ptr += 50
+            yield out[np.newaxis]
+
+    def preprocess(self, input_file):
+        """
+        input_file: str, file path
+        return: iterator
+        """
+        import ffmpeg
```
Comment (huangjun12) [inline tools/utils.py:550]: requirement.txt里面补充一下 ffmpeg，以及对应版本

**Review by huangjun12 — COMMENTED** [review]
commit时pre-commit检查一下格式

**Comment by huangjun12** [comment]
LGTM
